Internal/Deploy.ps1

```diff
@@ -15,323 +16,197 @@ $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-S
-    }
-    else {
-        $algoBranch = invoke-git -returnValue branch --show-current
-        Write-Host "Source branch: $algoBranch"
-    }
-    if ($collect) {
-        $status = invoke-git -returnValue status --porcelain=v1 | Where-Object { ($_) -and ($_.SubString(3) -notlike "Internal/*") }
-        if ($status) {
-            throw "Destination repo is not clean, cannot collect changes into dirty repo"
-        }
-    }
+    # Get Source Branch
+    $algoBranch = invoke-git -returnValue branch --show-current
+    Write-Host "Source branch: $algoBranch"
 
+    # Calculate Source SHA + Source Owner+Repo
+    $srcSHA = invoke-git -returnValue rev-parse HEAD
     $srcUrl = invoke-git -returnValue config --get remote.origin.url
-    if ($srcUrl.EndsWith('.git')) { $srcUrl = $srcUrl.Substring(0,$srcUrl.Length-4) }
+    if ($srcUrl.EndsWith('.git')) { $srcUrl = $srcUrl.Substring(0, $srcUrl.Length - 4) }
     $uri = [Uri]::new($srcUrl)
     $srcOwnerAndRepo = $uri.LocalPath.Trim('/')
+    Write-Host "Source SHA: $srcSHA"
     Write-Host "Source Owner+Repo: $srcOwnerAndRepo"
 
-    if (($config.PSObject.Properties.Name -eq "baseFolder") -and ($config.baseFolder)) {
-        $baseFolder =  Join-Path $config.baseFolder $config.localFolder 
-    }else {
-        $baseFolder = Join-Path ([Environment]::GetFolderPath("MyDocuments")) $config.localFolder
-    }
-
-    $copyToMain = $false
-    if ($config.PSObject.Properties.Name -eq "copyToMain") {
-        $copyToMain = $config.copyToMain
-    }
+    # baseFolder is the location in which we are going to clone AL-Go-Actions, AL-Go-PTE and AL-Go-AppSource
+    $baseFolder = Join-Path ([System.IO.Path]::GetTempPath()) ([GUID]::NewGuid().ToString())
+    New-Item $baseFolder -ItemType Directory | Out-Null
 
-    if (!(Test-Path $baseFolder)) {
-        New-Item $baseFolder -ItemType Directory | Out-Null
-    }
-    Set-Location $baseFolder
-
-    $config.actionsRepo, $config.perTenantExtensionRepo, $config.appSourceAppRepo | ForEach-Object {
-        if (Test-Path $_) {
-            Set-Location $_
-            if ($collect) {
-                $expectedUrl = "https://github.com/$($config.githubOwner)/$_.git"
-                $actualUrl = invoke-git -returnValue config --get remote.origin.url
-                if ($expectedUrl -ne $actualUrl) {
-                    throw "unexpected git repo - was $actualUrl, expected $expectedUrl"
-                }
-            }
-            else {
-                if (Test-Path ".git") {
-                    $status = invoke-git -returnValue status --porcelain
-                    if ($status) {
-                        throw "Git repo $_ is not clean, please resolve manually"
-                    }
-                }
-            }
-            Set-Location $baseFolder
-        }
+    # CopyToMain is set when we release to f.ex. v3.2, where we create a new branch on AL-Go-PTE called v3.2, but also copies the changes to the main and the preview branch

```

Comment:
Shouldn't `CopyToMain` always be set to _true_ if the target branch isn't _preview_ or _main_? Actually do we deploy to main? 🤔 

What are the scenarios?
1. Deploy to _preview_:
Deploys the AL-Go@preview to AL-Go-Actions@preview and AL-Go-PTE@preview and changes the SHA of the actions in AL-Go-PTE@preview to the latest SHA in AL-Go-Actions@preview.
2. Deploy a new version:
Deploys AL-Go@preview  to branches vX.Y in AL-Go-Actions and AL-Go-PTE and makes AL-Go-PTE@vX.Y point to AL-Go-Actions@vX.Y

I'm just asking if scenario 2. should include scenario 1.

Reply:
scenario 2 should include 1 AND deploy to AL-Go-PTE@main (i.e. copyToMain = Y)
People will in 99% of the time use AL-Go-PTE@main or AL-Go-PTE@preview as their templateUrl - not vX.Y